app/controllers/application_controller.rb

```diff
@@ -18,12 +20,27 @@ def sign_in(login_userinfo)
     user = User.user_from_userinfo(login_userinfo)
 
     @current_user = user
+    renew_session
     session[:userinfo] = login_userinfo
   end
 
   def sign_out
     @current_user = nil
     session.delete(:userinfo)
+    session.delete(:session_timeout_at)
+  end
+
+  def renew_session
+    session[:session_timeout_at] = Time.current + ENV.fetch("SESSION_TIMEOUT_IN_MINUTES", 15).to_i.minutes
+  end
+
+  def check_session_expiration
+    if logged_in? && (session[:session_timeout_at].blank? || session[:session_timeout_at] < Time.current)
```

Comment:
can it early return when the user is logged out, just to make this logic more readable?
```suggestion
    return unless logged_in?
    
    # user is logged_in
    if session[:session_timeout_at].blank? || session[:session_timeout_at] < Time.current
```